### PR TITLE
feat: implement simple routine and deploy calls

### LINKED ISSUE #26: Not allowing user to pick between .call() and .send() when doing routines.endPoint()
## Description

Presently, developers have the option to choose between two methods:

```javascript
routines.SomeEndpoint.call()
// or
routines.SomeEndpoint.send()
```

This choice may cause confusion, especially for new developers. Additionally, it introduces the risk of developers unintentionally calling a read-only endpoint using .send(), which is not only inefficient (consumes gas and creates a tesseract) but also considered bad practice.

The desired behavior is to have the js-moi-sdk automatically determine the appropriate method based on the nature of the endpoint being called. 

Additionally, if a developer intends to simulate interaction with a mutating function, they should continue to use routines.someEndpoint.call() (as per the existing convention). 

However, the option of using routines.someEndpoint.send() for this purpose should be deprecated and ultimately removed - The SDK should only decide if the endpoint being called is a mutating function and internally use .send() for such interactions.

## Possible resolution
To achieve this behavior, the js-moi-sdk should emulate the pattern used by ethers.js when handling different types of functions:

In case of an error when doing a mutating call: SDK should throw an error.

How ethers handle a mutate function call -  
```javascript
const tx = await erc20Contract.transfer(toAddress, amount);
const receipt = await tx.wait();
```

How ethers handle a view function call - 
```javascript
const balance = await erc20Contract.balanceOf(targetAddress);
```

### PR DESCRIPTION
**Pull request by sarvalabs-harshrastogi**
This pull request simplifies the logic routine calls and logic deployment by passing the ix option as the pass optional parameter. These optional parameters are `fuelPrice` and `fuelLimit` which are included in ix options. Internally it uses the `call` or `send` methods for which the above params are required. If the `fuelPrice` and `fuelLimit` are not explicitly passed. A request is sent to the `moi.FuelEstimate` method to get the fuel estimate and `fuel price` is default marked to `1`.

- closes #26 #29 #27 #28

**Note**: This is a breaking change.

### Changes include
- [ ] Bugfix (non-breaking change that solves an issue)
- [X] New feature (non-breaking change that adds functionality)
- [X] Breaking change (change that is not backwards-compatible and/or changes current functionality)

### Checklist
- [X] I have assigned this PR to myself
- [X] I have added at least 1 reviewer
- [X] I have tested this code
- [X] I have updated the README and other relevant documents (guides...)

### CODE REVIEW

File: packages/js-moi-logic/src/logic-factory.ts [hunk 1]
```diff
@@ -102,15 +102,15 @@ export class LogicFactory extends LogicBase {
 
         if(builder) {
             const builderRoutine = builder.data as LogicManifest.Routine;
-
-            if(builderRoutine.accepts && Object.keys(builderRoutine.accepts).length != args.length) {
+            
+            if(builderRoutine.accepts && (args.length < Object.keys(builderRoutine.accepts).length)) {
```
Comment (sarvalabs-gokul) [inline packages/js-moi-logic/src/logic-factory.ts:106]: don't not include optional params while checking for the expected param length.

File: packages/js-moi-logic/__tests__/logic.test.ts [hunk 2]
```diff
@@ -1,13 +1,12 @@
-import erc20 from "../manifests/erc20.json";
-import {JsonRpcProvider} from "js-moi-providers";
-import {LogicFactory} from "../src/logic-factory";
-import { getLogicDriver } from "../src/logic-driver";
-import { initializeWallet } from "./utils/utils";
+import { JsonRpcProvider } from "js-moi-providers";
 import { Signer } from "js-moi-signer";
 import { Wallet } from "js-moi-wallet";
+import todo from "../manifests/todo.json";
```
Comment (sarvalabs-gokul) [inline packages/js-moi-logic/__tests__/logic.test.ts:4]: replace the todo logic with erc20 logic.

File: packages/js-moi-logic/__tests__/utils/utils.ts [hunk 3]
```diff
@@ -1,11 +1,15 @@
+import { BaseProvider, JsonRpcProvider, VoyageProvider } from "js-moi-providers";
 import { Signer } from "js-moi-signer";
 import { Wallet } from "js-moi-wallet";
-import { JsonRpcProvider } from "js-moi-providers";
 
-export const initializeWallet = async (provider: JsonRpcProvider, mnemonic: string): Promise<Signer> => {
-    const derivationPath = "m/44'/6174'/0'/0/1";
+export const initializeWallet = async (provider: BaseProvider, mnemonic: string): Promise<Signer> => {
```
Comment (sarvalabs-gokul) [inline packages/js-moi-logic/__tests__/utils/utils.ts:5]: Use `JsonRPCProvider` for testing.

**Review by sarvalabs-gokul — CHANGES_REQUESTED** [review]

File: packages/js-moi-logic/__tests__/logic.test.ts (hunk 2 above)
Comment (sarvalabs-harshrastogi) [reply packages/js-moi-logic/__tests__/logic.test.ts]: a148b25d7be6dd2e55ef9f4502f56ebc49018142

File: packages/js-moi-logic/__tests__/utils/utils.ts (hunk 3 above)
Comment (sarvalabs-harshrastogi) [reply packages/js-moi-logic/__tests__/utils/utils.ts]: 1f3b5e6479daa109ff17ab3bc7100926914b4747

File: packages/js-moi-logic/src/logic-factory.ts (hunk 1 above)
Comment (sarvalabs-harshrastogi) [reply packages/js-moi-logic/src/logic-factory.ts]: a148b25d7be6dd2e55ef9f4502f56ebc49018142

**Pull request closed without merging**